### PR TITLE
add button to insert code to script

### LINKED ISSUE #22: Add button to insert code in RStudio script
## Prework

* [x] I understand and agree to `targetsketch`' [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to `targetsketch`' [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Proposal

Please describe the new feature. If applicable, write a minimal example in R code or pseudo-code to show input, usage, and desired output. To help us read any code you include (optional) please try to follow the [tidyverse style guide](https://style.tidyverse.org/). The `style_text()` and `style_file()` functions from the [`styler`](https://github.com/r-lib/styler) package make it easier.

Similar to [esquisse](https://github.com/dreamRs/esquisse/blob/master/R/module-controls.R), add insert code in script link or button.

### PR DESCRIPTION
**Pull request by parmsam**
# Prework

* [x] I understand and agree to this repository's [code of conduct](https://github.com/wlandau/targetsketch/blob/main/CODE_OF_CONDUCT.md).
* [x] I understand and agree to this repository's [contributing guidelines](https://github.com/wlandau/targetsketch/blob/main/CONTRIBUTING.md).
* [x] I have already submitted an issue to the [issue tracker](http://github.com/wlandau/targetsketch/issues) to discuss my idea with the maintainer.

# Summary

Please explain the purpose and scope of your contribution.

- Adds button to insert code in script
- Fixes #22 

# Related GitHub issues and pull requests

* Ref: #22 

# Checklist

* [x] This pull request is not a [draft](https://github.blog/2019-02-14-introducing-draft-pull-requests).

### CODE REVIEW

File: R/ui.R [hunk 1]
```diff
@@ -43,6 +43,15 @@ ui <- function() {
...
                 placement = "bottom"
               )
+            ),
+            shiny::div(
+              style = "display:inline-block",
+              shiny::actionButton(
+                inputId = "insert_code",
+                label = shiny::tagList(
+                  shiny::icon("arrow-circle-left"),
+                  "Insert code in script")
```
Comment (wlandau) [inline R/ui.R:53]: A minor thing, but would you move the end paren to the next line and use the same indent as `label = ...`?

File: R/server.R [hunk 2]
```diff
@@ -93,6 +93,13 @@ server <- function(input, output, session) {
       )
     }
   })
+  shiny::observeEvent(input$insert_code, {
+    insert_in_source_editor(input$script)
+    shinyalert::shinyalert(
+      title = "Inserted code into RStudio script",
+      type = "success"
+      )
```
Comment (wlandau) [inline R/server.R:101]: Very minor, but would you remove 2 spaces from the indent of the end paren?

**Review by wlandau — COMMENTED** [review]
This is a neat idea. I didn't fully understand when you posted #22, but now I get it after having run your fork/branch locally. I think I understand the intent, but I wonder about the direction. I will comment directly in #22.

File: R/server.R (hunk 2 above)
Comment (parmsam) [reply R/server.R]: Sorry about that. It defaulted to that position. Fixed.

File: R/ui.R (hunk 1 above)
Comment (parmsam) [reply R/ui.R]: Resolved. Thanks!

**Comment by wlandau** [comment]
Superseded by https://github.com/wlandau/targetsketch/pull/26

**Pull request closed without merging**